shoulder/register.py

```diff
@@ -0,0 +1,64 @@
+#
+# Shoulder
+# Copyright (C) 2018 Assured Information Security, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+class Register(object):
+    """ Models a register in the ARM architecture """
+    def __init__(self):
+        self.name = None
+        self.long_name = None
+        self.purpose = None
+        self.size = None
+        self.is_sysreg = True
+        self.fieldsets = []
+
+    def __str__(self):
+        msg = str(self.name) + " (" + str(self.long_name) + ")"
+        if self.purpose:
+            msg += ": " + str(self.purpose)
+        msg += "\nSize: " + str(self.size)
+        msg += ", System Register: " + str(self.is_sysreg)
```

Comment:
This would probably be more readable using [`str.format`](https://docs.python.org/3.1/library/functions.html#format), though I don't think I care enough to reject for that.

```python
"{name} ({long_name}): {purpose}\nSize: {size}, System Register: {is_sysreg}".format(
name=self.name, long_name=self.long_name,
purpose=self.purpose if self.purpose else "unspecified",
size=self.size,
is_sysreg=self.is_sysreg)
```

If you didn't have the funniness with `purpose` you could even eliminate the repetition with:

```python
"{name} ({long_name}): {purpose}\nSize: {size}, System Register: {is_sysreg}".format(**self.__dict__)
```

Reply:
I agree with using str.format, I don't know why but my first instinct is always to *not* use that whenever I print stuff with Python. Will update this one as well.